File: PHP/1080.php
<?php

class MaiorValor{

	private $index = -1;
	private $maiorValor = -1;

	function setter($index, $valor){
		$this->index = $index;
		$this->maiorValor = $valor;
	}

	function getter(){
		return [$this->index, $this->maiorValor];
	}

	function printer(){
		$aux = $this->getter();
		$index = $aux[0];
		$maiorValor = $aux[1];
		echo("{$maiorValor}\n{$index}\n");
	}

	function reader(){
		for ($i = 1; $i < 101; $i++){
			$aux = $this->getter();
			$index = $aux[0];
			$maiorValor = $aux[1];
			$n = (int) readline();
			if ($n > $maiorValor){
				$this->setter($i, $n);
			}
		}
	}
}

$maiorValor = new MaiorValor();
$maiorValor->reader();
$maiorValor->printer();
?>

File: Python/1080.py
class MaiorValor:
    def __init__(self):
        self.__index = -1
        self.__maior_valor = -1

    def setter(self, index: int, valor: int):
        self.__index = index
        self.__maior_valor = valor

    def getter(self) -> int:
        return self.__index, self.__maior_valor

    def printer(self):
        index, maior_valor = self.getter()
        print(f'{maior_valor}\n{index}')
    
    def reader(self):
        for i in range(1, 101):
            index, maior_valor = self.getter()
            n = int(input())
            if n > maior_valor:
                self.setter(i, n)

if __name__ == '__main__':
    mv = MaiorValor()
    mv.reader()
    mv.printer()
